### PR TITLE
PHP 8.1 Support - Phase 1

### PR DESCRIPTION
**Pull request by efrazier**
**Magento Phase 1 Testing / Summary**
**Tested on** https://magento.opensourcebrokers.ca/ 
**Admin:** eric@opensourcebrokers.ca : KxvVZd2a5p6VJwt3
**Magento:** 2.4.6 
**PHP:** PHP 8.1.23 
**Github Fork:** https://github.com/efrazier/integration-magento-osb
**FPD Client:** data.magento.opensourcebrokers.ca
**FPD Server:** magentos.opensourcebrokers.ca
**Test Summary:** [Google Doc Tests](https://docs.google.com/document/d/1TBKkOEcOKn52kSzkq6bCMM9Iw9hq4rwN1QGRExq9yFw/edit)

### CODE REVIEW

File: Block/Tealium.php [hunk 1]
```diff
@@ -209,33 +228,65 @@ public function render($type = null, $external = false, $sync = "sync")
...
+            
+            $this->fpurl = trim($this->fpurl);
+            if (!empty($this->fpurl) && $this->fpurl != "") {
+                // create the entire script tag to render for utag_data
+                $udo = <<<EOD
+                <!-- Tealium Universal Data Object / Data Layer -->
+                <div class="utagLib" style="display:none;">https://$this->fpurl/$this->profile/$this->target/utag.js</div>
+                <script type="text/javascript">
+                $udoJs
+                console.log(window);
```
Comment (corbinspicer68) [inline Block/Tealium.php]: Remove console.log

File: Block/Tealium.php [hunk 2]
```diff
@@ -209,33 +228,65 @@ public function render($type = null, $external = false, $sync = "sync")
...
+                <!-- ****************************************** -->
+                EOD;
+            } else {
+                // create the entire script tag to render for utag_data
+                $udo = <<<EOD
+                <!-- Tealium Universal Data Object / Data Layer -->
+                <div class="utagLib" style="display:none;">//tags.tiqcdn.com/utag/$this->account/$this->profile/$this->target/utag.js</div>
+                <script type="text/javascript">
+                $udoJs
+                console.log(window);
```
Comment (corbinspicer68) [inline Block/Tealium.php]: Remove console.log

File: view/frontend/web/js/utagLib.js [hunk 3]
```diff
@@ -30,39 +19,58 @@ define([
 					if (dataObject['cart_total_value'].length > 0) {
 						utag_data.cart_total_value = dataObject['cart_total_value'];
             		}
+
+					
 					console.log(utag_data);
-					console.log('bbb utag lib');
-				  	var LibURL = $.trim($('.utagLib').html());
+					console.log('Before utag lib');
```
Comment (corbinspicer68) [inline view/frontend/web/js/utagLib.js]: Remove console.log

File: view/frontend/web/js/utagLib.js [hunk 4]
```diff
@@ -30,39 +19,58 @@ define([
...
+						var LibURL = jQuery.trim(jQuery('.utagLib').html());
 						(function(a,b,c,d){
 								a=LibURL;
 								b=document;c='script';d=b.createElement(c);d.src=a;d.type='text/java'+c; 
 								d.async=true;
 								a=b.getElementsByTagName(c)[0];a.parentNode.insertBefore(d,a);
 							})();
+
+
+							/*
```
Comment (corbinspicer68) [inline view/frontend/web/js/utagLib.js]: Remove comment block

File: view/frontend/web/js/utagLib.js [hunk 5]
```diff
@@ -30,39 +19,58 @@ define([
...
-						utag_data.cart_total_value = dataObject['cart_total_value'];
-            		}
-					console.log(utag_data);
-		console.log('bbb utag lib');
-		});
-		*/
+	
 		console.log(utag_data);
-		console.log('aaaaa utag lib');
+		console.log('After utag lib');
```
Comment (corbinspicer68) [inline view/frontend/web/js/utagLib.js]: Remove console.log

File: Block/Tealium.php [hunk 6]
```diff
@@ -209,33 +228,65 @@ public function render($type = null, $external = false, $sync = "sync")
...
-</script>
-<!-- ****************************************** -->
-EOD;
+            
+            $this->fpurl = trim($this->fpurl);
+            if (!empty($this->fpurl) && $this->fpurl != "") {
+                // create the entire script tag to render for utag_data
+                $udo = <<<EOD
+                <!-- Tealium Universal Data Object / Data Layer -->
+                <div class="utagLib" style="display:none;">https://$this->fpurl/$this->profile/$this->target/utag.js</div>
```
Comment (corbinspicer68) [inline Block/Tealium.php]: There must be a better way of storing the loaded URL than including a hidden div?

File: view/frontend/web/js/utagLib.js [hunk 7]
```diff
@@ -30,39 +19,58 @@ define([
 					if (dataObject['cart_total_value'].length > 0) {
 						utag_data.cart_total_value = dataObject['cart_total_value'];
             		}
+
+					
 					console.log(utag_data);
```
Comment (corbinspicer68) [inline view/frontend/web/js/utagLib.js]: Remove console.log

**Review by corbinspicer68 — CHANGES_REQUESTED** [review]
Please correct the small changes.

File: Block/Tealium.php (hunk 6 above)
Comment (efrazier) [reply Block/Tealium.php]: Changed to JS var

File: Block/Tealium.php (hunk 1 above)
Comment (efrazier) [reply Block/Tealium.php]: removed

**Review by corbinspicer68 — APPROVED** [review]

**Review by corbinspicer68 — APPROVED** [review]

**Review by corbinspicer68 — APPROVED** [review]
Approved.

**Review by corbinspicer68 — DISMISSED** [review]

**Pull request closed without merging**